scripts/svm/closeDataWorkerLookUpTables.ts

```diff
@@ -0,0 +1,71 @@
+import * as anchor from "@coral-xyz/anchor";
+import { PublicKey, SendTransactionError } from "@solana/web3.js";
+import { AddressLookupTableProgram } from "@solana/web3.js";
+import yargs from "yargs";
+import { hideBin } from "yargs/helpers";
+
+// Set up the provider
+const provider = anchor.AnchorProvider.env();
+anchor.setProvider(provider);
+const connection = provider.connection;
+const payer = provider.wallet.publicKey;
+
+// Parse arguments
+// const argv = yargs(hideBin(process.argv))
+//   .option("payer", { type: "string", default: payer.toBase58(), describe: "Payer public key" }).argv;
+
+async function closeDataWorkerLookUpTables(): Promise<void> {
+  console.log("Starting process to close dataworker lookup tables...");
+  console.log(`Signer: ${payer.toBase58()}`);
+
+  try {
+    const accounts = await connection.getProgramAccounts(AddressLookupTableProgram.programId, {
+      filters: [{ memcmp: { offset: 22, bytes: payer.toBase58() } }],
+    });
+
+    console.log(`Found ${accounts.length} lookup tables.`);
+
+    for (const account of accounts) {
+      const lookupTableAddress = account.pubkey;
+      console.log(`\nProcessing lookup table: ${lookupTableAddress.toBase58()}`);
+
+      try {
+        // Attempt to deactivate the lookup table
+        const deactivateInstruction = AddressLookupTableProgram.deactivateLookupTable({
+          lookupTable: lookupTableAddress,
+          authority: payer,
+        });
+
+        const deactivateTransaction = new anchor.web3.Transaction().add(deactivateInstruction);
+        const deactivateSignature = await provider.sendAndConfirm(deactivateTransaction, [
+          (anchor.AnchorProvider.env().wallet as anchor.Wallet).payer,
+        ]);
+        console.log(`Deactivated: ${lookupTableAddress.toBase58()} | Tx Hash: ${deactivateSignature}`);
+      } catch {
+        console.error(`Pending deactivation: ${lookupTableAddress.toBase58()}`);
+      }
+
+      try {
+        // Attempt to close the lookup table
```

Comment:
Is this expected behavior I got that on the first run all my lookup tables get deactivated, but account closure fails. Once I rerun the script, deactivation fails, but account gets closed successfully?

Reply:
maybe there should be some await before trying to close account after the deactivation?

Reply:
there is a cooldown period after deactivation. after they are de-activated, only then can they be closed.

it's not posible, at least wit the limited research that I did, to detect this middle phase. we might be able to catch the errors better and enfer the status but for now the above logic achieved this without over engineering the flow.